### ISSUE TITLE
`with_collection` fails validation without specifying `with_collection_parameter`

### ISSUE THREAD

**Issue by reedrolemodel** [palkan/view_component-contrib#32]
## What did you do?

Generated a component named Project

## What did you expect to happen?

Render a collection using `Project::Component.with_collection(@projects))`

## What actually happened?

Error message:
```
The initializer for Project::Component doesn't accept the parameter `component`, which is required in order to render it as a collection.
To fix this issue, update the initializer to accept `component`.
See https://viewcomponent.org/guide/collections.html for more information on rendering collections.
```

## Additional context

https://github.com/viewcomponent/view_component/blob/main/lib/view_component/base.rb#L535 validates the component parameter by reflecting on the component class name (using `name.demodulize.underscore.chomp("_component").to_sym`). Because the generated component is called `Project::Component` it thinks the component param is `component` instead of `project`.

A workaround is adding `with_collection_parameter :project` to `Project::Component` but I was wondering if there was a better solution.

## Environment

**Ruby Version:** 3.2.2

**Framework Version (Rails, whatever):** 7.0.4.3

**View Component Contrib Version:** 0.1.3

**Comment by palkan** [comment]
I think, we should `with_collection_parameter :<component_name>` for every class inherited from `ViewComponentContrib::Base` if it follows the `<Name>::Component` pattern.

**Comment by reedrolemodel** [comment]
Yes, that is something that could easily be added to the `component.rb.tt` generator template. When I created this issue I was following [the How-to guide on Collections](https://viewcomponent.org/guide/collections.html) but have since moved on. It doesn't seem particularly useful now, but it was a blocker when getting started. For reference, it's the difference of
```diff
-    - projects.each do |project|
-      = component 'project/card', project:
+    = render(Project::Card::Component.with_collection(projects))
```
and
```diff
 class Project::Card::Component < ApplicationViewComponent
+  with_collection_parameter :project
```
which is not worth it in my opinion.

The issue I ran into could be solved by ViewComponent not insisting on the `Component` suffix, but that [has already been decided against](https://github.com/ViewComponent/view_component/blob/v3.0.0/docs/adrs/0002-naming-conventions-for-view-components.md).

It may be better to address this in the README saying something to the effect that naming opinions differ and suggest editing the generator template as a workaround.
